### DISCUSSION TITLE
Let [slug] render root of dynamic route e.g example.com/route vs only example.com/route/[slug]

### DISCUSSION THREAD

**Discussion by MariuzM** [vercel/next.js#19176] [Ideas]
# Feature request
When using [slug] for dynamic routes currently (or either i don't know) it only uses that component and renders everything after `/route/`. e.g. 

`example.com/route/page1` 
`example.com/route/page2`
`example.com/route/page3` 

I would like the `[slug].tsx` file to also have the ability to render `example.com/route/` when no `index.tsx` is present. 

## Describe the solution you'd like
Example i have: 

```
├── pages
│   ├── router
│   │   ├── [dynamicRoute].tsx
│   │   └── index.tsx
│   ├── _app.tsx
│   ├── _document.tsx
```

I have a folder `router` with file `index.tsx` that creates a route to `example.com/router` and it will render anything that is in `index.tsx`, the file contains a menu that has routes to different menu items. 

When i click on other items it renders them from `[dynamicRoute].tsx` which is what i want. But as soon as i hit route `example.com/router` is start re-rendering everything and uses `index.tsx` 

I would like to see the ability to render `example.com/router` but using [dynamicRoute].tsx because in that file i have conditions depending on what route it will render specific component, without causing page reload / refresh / component reload.

Attaching a video that shows when clicking `/router` button it re-direct to `example.com/router` thus it starts using `index.tsx` and re-renders the hole component wheres compared to `[dynamicRoute].tsx` it only re-renders that specific element. 
https://www.dropbox.com/s/5kuxg784s78ndy9/Screen%20Recording%202020-11-14%20at%2021.58.03.mov?dl=0

[dynamicRoute].tsx
```tsx
import { Button, Heading } from '@chakra-ui/react'
import { useRouter } from 'next/router'

const DynamicRoute = () => {
  const router = useRouter()
  console.log('DynamicRoute -> router', router.asPath)

  return (
    <>
      <Heading>[dynamicRoute].tsx File</Heading>
      <Button onClick={() => router.push('/router')}>/router</Button>
      <Button onClick={() => router.push('page1')}>/router/page1</Button>
      <Button onClick={() => router.push('page2')}>/router/page2</Button>
      <Button onClick={() => router.push('page3')}>/router/page3</Button>

      {router.asPath === '/router/' && <div>/router</div>}
      {router.asPath === '/router/page1' && <div>/router/page1</div>}
      {router.asPath === '/router/page2' && <div>/router/page2</div>}
      {router.asPath === '/router/page3' && <div>/router/page3</div>}
    </>
  )
}

export default DynamicRoute
```

index.tsx
```tsx
import { useRouter } from 'next/router'
import { Button, Heading } from '@chakra-ui/react'

const DrRoot = () => {
  const router = useRouter()

  return (
    <>
      <Heading>Index.tsx File</Heading>
      <Button onClick={() => router.push('/')}>Home</Button>
      <Button onClick={() => router.push(`${router.pathname}/page1`)}>Page 1 Button</Button>
      <Button onClick={() => router.push(`${router.pathname}/page2`)}>Page 2 Button</Button>
      <Button onClick={() => router.push(`${router.pathname}/page3`)}>Page 3 Button</Button>
    </>
  )
}

export default DrRoot
```

To recap, i just want this component to be rendered. 
![Screen Shot 2020-11-14 at 22 30 37](https://user-images.githubusercontent.com/36937039/99158287-1b1b7b80-26c9-11eb-95b7-342f1b4bc168.png)

**Comment by MariuzM** [comment]
So thinking the scenario could be. 

If `[dynamicRoute]` is in a folder and there is no `index` file then `[dynamicRoute]` file should also act like `index` file

So then i can have a condition like this `{router.asPath === '/router' && <div>/router</div>}` without re-rendering the hole component.

**Comment by eyenalxai** [comment]
Yes please! Struggling with it right now 😔
